rust/tool/src/utils.rs

```diff
@@ -0,0 +1,55 @@
+use std::fs;
+use std::io::Write;
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Result};
+use tempfile::TempDir;
+
+/// A temporary directory that enables creating secure temporary files in it.
+///
+/// The temporary directory has three useful properties:
+///
+/// - Files are created with mode 0o600, so that they are only accessible by the current user.
+/// - Files are named and not entirely ephemeral (unlike a real temporary file).
+/// - The directory and its children are cleaned up (i.e. deleted) when the variable that holds the
+///   directory goes out of scope.
+///
+/// This lightly protects against an attacker without root access from modifying files undetected.
+/// Additionally, because the files have named paths, they can be passed to external programs while
+/// still being securely deleted after they are not needed anymore.
+pub struct SecureTempDir(TempDir);
```

Comment:
nit: You are extending a type with some methods. The typical way in Rust to achieve this is not to wrap the type in a struct, because this will lead to "pass-through" implementations of some functions (your `path` is a case in point). Instead you define a trait with your new functionality and implement it for `TempDir`: https://cmcenroe.me/2016/08/22/rust-extending-type.html

Reply:
Thanks! I turned it into an extension trait.